repository/firestore/firestore_test.go

```diff
@@ -83,13 +83,71 @@ func (suite *FirestoreRepositoryTestSuite) TestCreateBulk() {
 	suite.Assert().ErrorIs(err, repository.ErrMethodNotImplemented)
 }
 
-func (suite *FirestoreRepositoryTestSuite) TestFind() {
-	var all []Test
+func (suite *FirestoreRepositoryTestSuite) TestFind_All() {
+	var found []Test
 
 	suite.setupMockData()
 
-	suite.Require().NoError(suite.repository.Find(&all))
-	suite.Assert().Len(all, 3)
+	suite.Require().NoError(suite.repository.Find(&found))
+	suite.Assert().Len(found, 3)
+}
+
+func (suite *FirestoreRepositoryTestSuite) TestFind_MaxResults() {
+	var found []Test
+
+	suite.setupMockData()
+
+	// Calling with max results should return the same amount of elements
+	suite.Require().NoError(suite.repository.Find(&found, MaxResults(1)))
+	suite.Assert().Len(found, 1)
+}
+
+func (suite *FirestoreRepositoryTestSuite) TestFind_Offset() {
+	var found []Test
+
+	suite.setupMockData()
+
+	// Calling with offset should return total - offset elements.
+	suite.Require().NoError(suite.repository.Find(&found, Offset(1)))
+	suite.Assert().Len(found, 2)
```

Comment:
Can we verify that the expected number of records are being skipped?

Reply:
Unless you think we should use the `Order`, but that will make this test depend on a different feature.

Reply:
Thinking about this again, I suppose the check I proposed does not make sense in a NoSQL context.

Still, I would like to create one additional test that validates `Order` + `MaxResults` + `Offset`, mainly because it will enable pagination in the future.

Reply:
I've added a test case covering different pagination scenarios